### ISSUE TITLE
Unclear compiler error when it cannot infer higher-kinded type correctly

### ISSUE THREAD

**Issue by kpodsiad** [lampepfl/dotty#18129]
## Compiler version

3.3.0

## Minimized example

```Scala
//> using scala "3.3.0"
// > using scala "2.13.11" // << works fine

trait Typeclass[F[_]]
class Foo[F[_]]()

object Example {
  def makeWork[F[_]: Typeclass](someArg: Option[Foo[F]] = None): Option[Foo[F]] = ??? 
}

class Usage[F[_]: Typeclass] {
  val x1: Foo[F] = Example.makeWork[F]().get // << works correctly
  val x2: Foo[F] = Example.makeWork().get // << doesn't work, really unclear compiler error
}

```

## Output

```scala
missing argument for parameter evidence$2 of method makeWork in object Example: (implicit evidence$2: Typeclass[[_] =>> Any]): Option[Foo[[_] =>> Any]]
```

## Expectation

`[_] =>> Any` might be obvious for people used to Scala 3, but I'm not sure it's so obvious for other people. I was scratching my head for a while till I realized that what compiler is really trying to say is "I can't infer type properly, will you help me?". I'd message to look like `Cannot infer F type in method makeWork. Probably you need to make this type explicit`, basically anything that will inform user what's the issue is really about.

**Comment by prolativ** [comment]
In general it's not that simple for the compiler to tell how to fix a problem like that as type inference might influence the implicit search and vice versa. Independently from any improvements in the compiler, one can try to use `@implicitNotFound` annotation to customize the compiler's error messages. If there's a chance the compiler might not have enough information to correctly infer type parameters based on other method parameters (just like in the snippet above) it seems to be a good practice to add an extra contextual parameter to assert that the inferred type was not the most generic one that could be inferred (like `Nothing` or `Any` - although at the moment I'm not sure how to make this work for higher order types).

**Comment by prolativ** [comment]
Actually it seems like the inference issue is already fixed in 3.3.1-RC1 (also in 3.3.1-RC2 and the latest nightly)
